civis_jupyter_notebooks/assets/git-init

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# TODO: write conditional for setting up SSH Key
+
+if test $GIT_CREDENTIAL; then
+  git config --global credential.helper 'cache --timeout=10000000'
+  git credential approve <<EOF
+protocol=https
+host=github.com
+username=$GIT_CREDENTIAL
+password=x-oauth-basic
+EOF
+fi
+
+FULL_REPO_URI="https://$GIT_REPO_URL"
+if ! git clone "$FULL_REPO_URI" --depth=1 --branch "$REPO_REF"; then
```

Comment:
@salilgupta1 Is this sufficient if users are meant to be using the terminal to push?  I'm worried some users might be expecting features only available with a full clone.

Reply:
can you elaborate on what the differences are. I didn't spend too much time looking into this command - copied it over from our shiny work

Reply:
Jk I understand what this command is doing. This is a great question ... From a perf perspective it would be nice to limit how much we pull down. But a user would want the ability to view history. I think since this a development environment where you are actively changing code we can just do a full clone for now. But let's make a note and make sure this is reviewed before external.